src/custom/utils/metadata.ts

```diff
@@ -43,3 +51,18 @@ export function generateAppDataDoc(metadata: MetadataDoc = {}): AppDataDoc {
     },
   }
 }
+
+export async function uploadMetadataDocToIpfs(appDataDoc: AppDataDoc): Promise<string> {
```

Comment:
should we have a generic `api/ipfs` file with `uploadTextFileToIpfs`, with most of this logic so `uploadMetadataDocToIpfs` just serializes the `appDataDoc` in to a text and use this function? 

This way, we differenciate between the ipfs integration, and the utility for uploading metadata.